controller/helpers.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	cfglib "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/logp/configure"
+	"github.com/elastic/elastic-agent-libs/paths"
+	"github.com/elastic/elastic-agent-shipper/config"
+)
+
+// I am not net sure how this should work or what it should do, but we need to read from that error channel
```

Comment:
You should log the error, it's a way for the agent to report failures making gRPC calls to somewhere that can log them.

Reply:
Example where it is used https://github.com/elastic/elastic-agent-client/blob/4477e3ace394ef1abfec55afa5cc5e1f6f87980c/pkg/client/client_v2.go#L248-L252

Reply:
Also reading this I realized that Beats doesn't do this https://github.com/elastic/beats/issues/34381